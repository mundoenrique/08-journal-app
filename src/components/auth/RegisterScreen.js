export default function RegisterScreen() {
	return (
		<>
			<h1>RegisterScreen</h1>
		</>
	);
}
