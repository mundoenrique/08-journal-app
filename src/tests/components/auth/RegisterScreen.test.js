import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import RegisterScreen from '../../../components/auth/RegisterScreen';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
	auth: {},
	ui: {
		loading: false,
		msgError: null,
	},
	notes: {
		notes: [],
		active: null,
	},
};

let store = mockStore(initState);

const wrapper = mount(
	<Provider store={store}>
		<MemoryRouter>
			<RegisterScreen />
		</MemoryRouter>
	</Provider>
);

describe('Pruebas sobre <RegisterScreen/>', () => {
	test('Debe de mostrar el componente <RegisterScreen/>', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe hacer el dispatch a la acciÃ³n respectiva', () => {
		const inputEmail = wrapper.find('input[name="email"]');
		inputEmail.simulate('change', {
			target: {
				value: '',
				name: 'email',
			},
		});

		wrapper.find('form').simulate('submit', {
			preventDefault() {},
		});

		const actions = store.getActions();

		expect(actions[0]).toEqual({
			type: types.uiSetError,
			payload: 'Invalid email',
		});
	});

	test('Debe de mostrar la caja de alerta con el error', () => {
		const initState = {
			auth: {},
			ui: {
				loading: false,
				msgError: 'Email is required',
			},
			notes: {
				notes: [],
				active: null,
			},
		};

		const store = mockStore(initState);

		const wrapper = mount(
			<Provider store={store}>
				<MemoryRouter>
					<RegisterScreen />
				</MemoryRouter>
			</Provider>
		);

		expect(wrapper.find('.auth__alert-error').exists()).toBe(true);
		expect(wrapper.find('.auth__alert-error').text().trim()).toBe(
			'Email is required'
		);
	});
});
