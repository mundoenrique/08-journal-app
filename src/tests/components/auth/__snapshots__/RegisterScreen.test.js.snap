// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <RegisterScreen/> Debe de mostrar el componente <RegisterScreen/> 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Yajaira"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="yayita@gmail.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Password Confirmation"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Alredy have an account?
    </a>
  </form>,
]
`;
